contracts/libraries/CronUtility.sol

```diff
@@ -0,0 +1,476 @@
+   * only be called off-chain.
+   * @param spec the spec to evaluate
+   * @return the next tick
+   */
+  function nextTick(Spec memory spec) public view returns (uint256) {
+    uint16 year = DateTime.getYear(block.timestamp);
+    uint8 month = DateTime.getMonth(block.timestamp);
+    uint8 day = DateTime.getDay(block.timestamp);
+    uint8 hour = DateTime.getHour(block.timestamp);
+    uint8 minute = DateTime.getMinute(block.timestamp);
+    uint8 dayOfWeek;
+    for (; true; year++) {
+      for (; month <= 12; month++) {
+        if (!matches(month, spec.month)) {
+          day = 1;
+          hour = 0;
+          minute = 0;
+          continue;
+        }
+        for (; day <= DateTime.getDaysInMonth(month, year); day++) {
+          if (!matches(day, spec.day)) {
+            hour = 0;
+            minute = 0;
+            continue;
+          }
+          dayOfWeek = DateTime.getWeekday(
+            DateTime.toTimestamp(year, month, day)
+          );
+          if (!matches(dayOfWeek, spec.dayOfWeek)) {
+            hour = 0;
+            minute = 0;
+            continue;
+          }
+          for (; hour < 24; hour++) {
+            if (!matches(hour, spec.hour)) {
+              minute = 0;
+              continue;
+            }
+            for (; minute < 60; minute++) {
+              if (!matches(minute, spec.minute)) {
+                continue;
+              }
+              return DateTime.toTimestamp(year, month, day, hour, minute);
+            }
+            minute = 0;
+          }
+          hour = 0;
+        }
+        day = 1;
+      }
+      month = 1;
+    }
+  }
+
+  /**
+   * @notice matches evaluates if a values matches a field.
+   * ex: 3 matches *, 3 matches 0-5, 3 does not match 0,2,4
+   * @param value the value of a field
+   * @param field the field struct to match against
+   * @return true / false if they match
+   */
+  function matches(uint8 value, Field memory field)
+    private
+    pure
+    returns (bool)
+  {
+    if (field.fieldType == WILD) {
+      return true;
+    } else if (field.fieldType == INTERVAL) {
+      return value % field.interval == 0;
+    } else if (field.fieldType == EXACT) {
+      return value == field.singleValue;
+    } else if (field.fieldType == RANGE) {
+      return value >= field.rangeStart && value <= field.rangeEnd;
+    } else if (field.fieldType == LIST) {
+      for (uint256 idx = 0; idx < field.listLength; idx++) {
+        if (value == field.list[idx]) {
+          return true;
+        }
+      }
+      return false;
+    }
+    revert unknownFieldType();
+  }
+
+  // VALIDATIONS
+
+  /**
+   * @notice validate validates a spec, reverting if any errors are found
+   * @param spec the spec to validate
+   * @return the original spec
+   */
+  function validate(Spec memory spec) private pure returns (Spec memory) {
+    validateMinute(spec.minute);
+    validateField(spec.minute, MINUTE, 5, 0, 59);
```

Comment:
An interval with a shorter cadence has a greater chance of interfering with itself and causing failed txs / wasted money. Ex if a gas spike is happening and txs take longer than a minute to confirm.

Reply:
We could remove this restriction, but I'm less confident it will always behave as intended

Reply:
It seems like a little bit of a mixture of concerns. Ideally cron would know nothing about how Keepers behave and then Keeper would enforce this restriction. That might not be practical or gas efficient, but from an object-design/encapsulation perspective I think it would be best.

Reply:
I'm wondering if maybe I should just remove the minute validation all together. We can just warn customers that it might not work as well at such short intervals. But it's not really our problem if they choose to push the limits and they're paying for the services they receive.

Reply:
I think I'll do that for now, and we can add it back in on the upkeep side if we think it's necessary